vms/platformvm/genesis/genesistest/genesis.go

```diff
@@ -10,98 +10,166 @@ import (
+	DefaultTimeUnix             = uint64(DefaultTime.Unix())
+	DefaultValidatorEndTime     = DefaultTime.Add(DefaultValidatorDuration)
+	DefaultValidatorEndTimeUnix = uint64(DefaultValidatorEndTime.Unix())
 )
 
-func New(t testing.TB) *genesis.Genesis {
-	require := require.New(t)
+var (
+	// Keys that are funded in the genesis
+	DefaultFundedKeys = secp256k1.TestKeys()
 
-	genesisValidator := &txs.AddValidatorTx{
-		Validator: txs.Validator{
-			NodeID: ValidatorNodeID,
-			Start:  TimeUnix,
-			End:    ValidatorEndTimeUnix,
-			Wght:   ValidatorWeight,
-		},
-		StakeOuts: []*avax.TransferableOutput{
-			{
-				Asset: AVAXAsset,
-				Out: &secp256k1fx.TransferOutput{
-					Amt: ValidatorWeight,
-				},
-			},
-		},
-		RewardsOwner:     ValidatorRewardsOwner,
-		DelegationShares: ValidatorDelegationShares,
+	// Node IDs of genesis validators
+	DefaultNodeIDs []ids.NodeID
+)
+
+func init() {
+	DefaultNodeIDs = make([]ids.NodeID, len(DefaultFundedKeys))
+	for i := range DefaultFundedKeys {
+		DefaultNodeIDs[i] = ids.GenerateTestNodeID()
 	}
-	genesisValidatorTx := &txs.Tx{Unsigned: genesisValidator}
-	require.NoError(genesisValidatorTx.Initialize(txs.Codec))
+}
 
-	genesisChain := &txs.CreateChainTx{
-		SubnetID:   constants.PrimaryNetworkID,
-		ChainName:  XChainName,
-		VMID:       constants.AVMID,
-		SubnetAuth: &secp256k1fx.Input{},
+type Config struct {
+	NodeIDs          []ids.NodeID
+	ValidatorWeight  uint64
+	Time             time.Time
+	ValidatorEndTime time.Time
+
+	FundedKeys     []*secp256k1.PrivateKey
+	InitialBalance uint64
+}
+
+func New(t testing.TB, c Config) *platformvmgenesis.Genesis {
+	if len(c.NodeIDs) == 0 {
+		c.NodeIDs = DefaultNodeIDs
+	}
+	if c.ValidatorWeight == 0 {
+		c.ValidatorWeight = DefaultValidatorWeight
+	}
+	if c.Time.IsZero() {
+		c.Time = DefaultTime
+	}
+	if c.ValidatorEndTime.IsZero() {
+		c.ValidatorEndTime = DefaultValidatorEndTime
 	}
-	genesisChainTx := &txs.Tx{Unsigned: genesisChain}
-	require.NoError(genesisChainTx.Initialize(txs.Codec))
-
-	return &genesis.Genesis{
-		UTXOs: []*genesis.UTXO{
-			{
-				UTXO: avax.UTXO{
-					UTXOID: avax.UTXOID{
-						TxID:        AVAXAssetID,
-						OutputIndex: 0,
+	if len(c.FundedKeys) == 0 {
+		c.FundedKeys = DefaultFundedKeys
+	}
+	if c.InitialBalance == 0 {
+		c.InitialBalance = DefaultInitialBalance
+	}
+
+	require := require.New(t)
+
+	genesis := &platformvmgenesis.Genesis{
+		UTXOs:         make([]*platformvmgenesis.UTXO, len(c.FundedKeys)),
+		Validators:    make([]*txs.Tx, len(c.NodeIDs)),
+		Timestamp:     uint64(c.Time.Unix()),
+		InitialSupply: InitialSupply,
+	}
+	for i, key := range c.FundedKeys {
+		genesis.UTXOs[i] = &platformvmgenesis.UTXO{UTXO: avax.UTXO{
+			UTXOID: avax.UTXOID{
+				TxID:        snowtest.AVAXAssetID,
+				OutputIndex: uint32(i),
+			},
+			Asset: AVAXAsset,
+			Out: &secp256k1fx.TransferOutput{
+				Amt: c.InitialBalance,
+				OutputOwners: secp256k1fx.OutputOwners{
+					Threshold: 1,
```

Comment:
`Threshold` is the number of keys from `Addrs` that must come together in order to spend this output. Because there is 1 addr in the `Addrs` array, `Threshold` _must_ be `1`. Any other value would be invalid due to inefficiency or being unspendable.

Reply:
Yes that's correct.

The avalanchego wallet supports partial signing for this purpose. I actually don't know that I have a good docs pointer for this... I could put up a quick wallet example if you'd be interested.

Reply:
Put up an example here: #3325